gluten-core/src/main/scala/org/apache/spark/util/memory/TaskMemoryResources.scala

```diff
@@ -155,4 +160,29 @@ class TaskMemoryResourceRegistry extends Logging {
   private[memory] def getSharedMetrics(): TaskMemoryMetrics = {
     sharedMetrics
   }
+
+  private[memory] def addRecycler(recycler: TaskMemoryResourceRecycler) = {
```

Comment:
Why can't we use the addTaskCompletionListener?

Reply:
`addTaskCompletionListener` doesn't implement priority for listeners. In our case we should make sure the allocators /memorypools to be released after all dependent buffers are released.